### PR TITLE
Re-implementing Next State in Backbone

### PR DESCRIPTION
**Pull request by YesugenB**
Closes issue #542 and #549

### CODE REVIEW

**Comment by amgrubb** [comment]
Ok, let's "code freeze" this branch and start reviewing it.

**Review by Domily11 — DISMISSED** [review]

File: leaf-ui/js/backendComm.js [hunk 1]
```diff
@@ -121,29 +51,26 @@ function backendStringifyCirc(obj){
...
 function responseFunc(analysisRequest, response) {
-	$("body").removeClass("waiting"); //Remove spinner under cursor 
+	$("body").removeClass("spinning"); // Remove spinner from page
 	var results = JSON.parse(response);
 	if (errorExists(results)) {
 		var msg = getErrorMessage(results.errorMessage);
 		alert(msg);
 	} else {
-		console.log(analysisRequest.get('action'))
+		console.log(analysisRequest.get('action'));
```
Comment (Domily11) [inline leaf-ui/js/backendComm.js:60]: Remove console.log

File: leaf-ui/js/backendComm.js [hunk 2]
```diff
@@ -121,29 +51,26 @@ function backendStringifyCirc(obj){
...
+		console.log(JSON.stringify(results));
 		if (results == "") {
 			alert("Error while reading the response file from server. This can be due an error in executing java application.");
 			return;
+		} else if (analysisRequest.get('action') == 'singlePath' || analysisRequest.get('action') == 'updatePath') {
+			var analysisResult = convertToAnalysisResult(results); 	// {ResultBBM}
+			displayAnalysis(analysisResult, false);
+			console.log(analysisResult);
```
Comment (Domily11) [inline leaf-ui/js/backendComm.js]: Remove??

File: leaf-ui/js/onFunctions.js [hunk 3]
```diff
@@ -330,14 +483,35 @@ paper.on("link:options", function (cell) {
...
-        console.log(curRequest);
-        backendSimulationRequest(curRequest);    
+
+        // Checks to see if single path has been run by seeing if there are any results
+        if (typeof curRequest.previousAttributes().results === 'undefined' || curRequest.previousAttributes().results.length == 0) {
+             var singlePathRun = false;
+        } else {
+            var singlePathRun = true;
+        }
+        console.log(singlePathRun);
```
Comment (Domily11) [inline leaf-ui/js/onFunctions.js]: Remove?

File: leaf-ui/js/onFunctions.js [hunk 4]
```diff
@@ -384,6 +558,7 @@ paper.on("link:options", function (cell) {
      */
     function switchToAnalysisMode() {
         setInteraction(false);
+//        console.log(paper)
```
Comment (Domily11) [inline leaf-ui/js/onFunctions.js]: Remove

File: leaf-ui/js/onFunctionsNextState.js [hunk 5]
```diff
@@ -10,11 +10,80 @@ $('#btn-zoom-out').on('click', function () { zoomOut(analysis.paperScroller); })
...
+    pPaperScroller.zoom(-0.2, { min: 0.2 });
+}
+
+/**
+ * Helper function function for fontUp, fontDown, and defaultFont
+ * @param {int} new_font 
+ */
+function changeFont(new_font, pPaper) {
+    var elements = analysis.graph.getElements();
+    console.log(elements);
```
Comment (Domily11) [inline leaf-ui/js/onFunctionsNextState.js]: Remove

File: leaf-ui/rappid-extensions/BackboneModelsAnalysis.js [hunk 6]
```diff
@@ -46,10 +44,12 @@ var ResultBBM = Backbone.Model.extend({
             elementList: null,
             allSolutions: null, 
             nextStateTPs: null,
-            isPathSim: false, // Used for slider visualization  // Should be set to true for single path?
+//            isPathSim: false, // Used for slider visualization  // Should be set to true for single path?
```
Comment (Domily11) [inline leaf-ui/rappid-extensions/BackboneModelsAnalysis.js]: Remove if not needed

File: leaf-ui/js/backendComm.js (hunk 1 above)
Comment (amgrubb) [reply leaf-ui/js/backendComm.js:60]: We'll keep these for now.

File: leaf-ui/js/backendComm.js (hunk 2 above)
Comment (amgrubb) [reply leaf-ui/js/backendComm.js]: Yup, removed.

**Review by amgrubb — DISMISSED** [review]
Looks good, so far.

**Review by Domily11 — DISMISSED** [review]

**Review by Domily11 — APPROVED** [review]

**Pull request merged**